src/malli/generator.cljc

```diff
@@ -454,6 +454,22 @@
                         {:infinite? (get props :gen/infinite? false)
                          :NaN? (get props :gen/NaN? false)})
                       (-min-max schema options))))
+(defmethod -schema-generator :float [schema options]
+  (let [max-float #?(:clj Float/MAX_VALUE :cljs (.-MAX_VALUE js/Number))
+        min-float (- max-float)
+        ceil-max (fn [v] (if (nil? v) max-float (min max-float v)))
+        floor-min (fn [v] (if (nil? v) min-float (max min-float v)))
+        props (m/properties schema options)
+        min-max-props (-min-max schema options)
+        min-max-clamped #?(:clj {:min (floor-min (:min min-max-props))
+                                 :max (ceil-max (:max min-max-props))}
+                           :cljs min-max-props)
```

Comment:
I can't think of a compelling reason to clamp the bounds, perhaps an error should be thrown instead.

Relatedly, the bounds should be coerced to float so then floating point numbers are generated regardless of bounds like in this PR https://github.com/metosin/malli/pull/1052

Reply:
> I can't think of a compelling reason to clamp the bounds, perhaps an error should be thrown instead.

Well the main reason I thought to clamp the bounds is because we're using `gen/double*`, which can produce numbers that will lead to thrown exceptions when trying to coerce them into floats. So I think that means we always need to clamp the bounds, but maybe we can also throw an error if someone tries to override those boundaries (?).

> the bounds should be coerced to float so then floating point numbers are generated regardless of bounds

I'll push up a commit to coerce the bounds to floats too 👍